src/data/analytics.js

```diff
@@ -1,79 +1,97 @@
-export const choseMyCommitments    = () => trackMenuEvent('my commitments')
-
-export const filteredRequestsByDestination = query => trackRequestsEvent('destination', query)
-export const filteredRequestsByEvent       = name  => trackRequestsEvent('event', name)
-export const filteredRequestsByOrigin      = query => trackRequestsEvent('origin', query)
-export const filteredRequestsBySize        = size  => trackRequestsEvent('filter', size)
-export const filteredRequestsByMine        = ()    => trackRequestsEvent('filter', 'mine')
-export const filteredRequestsByProviding   = ()    => trackRequestsEvent('filter', 'providing')
-export const filteredRequestsByAll         = ()    => trackRequestsEvent('filter', 'all')
-export const viewedRequestsAs              = type  => trackRequestsEvent('view'  , type)
-export const searchedRequests              = query => trackRequestsEvent('search', query)
-
-export const filteredMeetingsByAll      = ()    => trackMeetingsEvent('filter', 'all')
-export const filteredMeetingsByLocation = query => trackMeetingsEvent('location', query)
-export const filteredMeetingsByMine     = ()    => trackMeetingsEvent('filter', 'mine')
-export const searchedMeetings           = query => trackMeetingsEvent('search', query)
-
-export const accepted       = () => trackRequestEvent('accepted')
-export const offered        = () => trackRequestEvent('offered')
-export const trackReceived  = () => trackRequestEvent('received')
-export const trackDelivered = () => trackRequestEvent('delivered')
-export const updated        = () => trackRequestEvent('updated')
-export const created        = () => trackRequestEvent('created')
-export const cancelled      = () => trackRequestEvent('cancelled')
-
-export const changedNickname = () => trackUserEvent('changed nickname')
-export const changedAvatar   = () => trackUserEvent('changed avatar')
-
-export const sentMessage = () => trackEvent('Messaging', 'sent')
-
-export const uploadedImage = () => trackEvent('Image', 'uploaded')
-
-export const createdWatch = () => trackEvent('watch', 'created')
+export const notFound = () => trackEvent('Error', 'Page not found')
+
+const trackAuthEvent     = eventParameters => trackEvent('Auth'    , eventParameters)
+const trackMeetingsEvent = eventParameters => trackEvent('Meetings', eventParameters)
+const trackMenuEvent     = eventParameters => trackEvent('Menu'    , eventParameters)
+const trackRequestsEvent = eventParameters => trackEvent('Requests', eventParameters)
+const trackRequestEvent  = eventParameters => trackEvent('Request' , eventParameters)
+const trackUserEvent     = eventParameters => trackEvent('User'    , eventParameters)
+
+export const loggedIn  = () => trackAuthEvent({event_label: 'logged in'})
+export const loggedOut = () => trackAuthEvent({event_label: 'logged out'})
+
+export const createRequestByFab    = () => trackMenuEvent({event_label: 'create request', value: 'fab'})
```

Comment:
Did you consider just using two string arguments, rather than an object argument that requires specifying the key names every time you call it? That could (for example) simplify this...
```js
function trackEvent(eventName, eventParameters) {
  gtag('event', eventName, eventParameters)
}

// ...

const trackMenuEvent = eventParameters => trackEvent('Menu' , eventParameters)

// ...

export const createRequestByFab = () => trackMenuEvent({event_label: 'create request', value: 'fab'})
```
... to this:
```js
function trackEvent(eventName, eventLabel, value) {
  gtag('event', eventName, { event_label: eventLabel, value: value })
}

// ...

const trackMenuEvent = eventParameters => trackEvent('Menu', eventLabel, value)

// ...

export const createRequestByFab = () => trackMenuEvent('create request', 'fab')
```

One result would be that most of this file would go back to being unchanged.

Reply:
No I hadn't, but that makes good sense.